src/app/components/Timer.js

```diff
@@ -0,0 +1,30 @@
+export const Timer = (duration) => {
```

Comment:
Cieszę się, że czujesz się na tyle dobrze! :) Brawo i to działa :) 
Z samego działania dobrze byłoby zatrzymać timer jak upłynie ten czas. Żeby już przestał odliczać. Bo tutaj masz tego ifa, a jednak dalej działa. 

Na razie mała podpowiedź. Zobaczymy :) 
Jasne, że testy są problemem. Każdy test który jest w jakiś sposób niedeterministyczny czy zależny od czasu jest problematyczny. Bo testy muszą być szybkie i jednoznaczne. Tutaj jak czekasz ileś to nie jest droga :) 

Więc tak spróbuj Timer nazwać TextTimer (bo mamy też jeszcze Lightsaber w przygotowaniu) i teraz nazwa troche nie będzie Ciebie mylić. Odpowiedzialnością TexTimer będzie po prostu wyświetlanie czasu (taka warstwa wyświetlania - mógłbyś równie dobrze właśnie zrobić to w konsoli czy coś) i teraz... spróbuj zrobić jakąś **abstrakcję** na upływ czasu. I żeby TextTimer był jej świadomy. W teście tą abstrakcję możesz np. zastąpić i wywołać upływ czasu (tak zfejkować, że czas upływa, a nie ralnie czekać). 

Pomóc Ci też może omówienie na filmiku od 22:44. Dotyczy miecza, ale porady są takie same jak dla textu. 
https://youtu.be/X71Slaok-fs?t=1364

Spróbuj pod spodem w komentarzu opisać swoimi słowami - co myślisz, że trzeba zrobić :)

Reply:
Ok, czyli chcielibyśmy, żeby odmierzanie czasu samo w sobie było niezależną funkcją, a jej 'wynik' byłby przekazywany do komponentów. Wtedy każdy z komponentów czy to miecz czy napis może wywołać tą funkcję dla siebie. Tak to sobie wyobrażam, nie wiem czy dobrze i nie jestem jeszcze pewny szczegółów implementacji :P

Reply:
No tak mniej więcej :) 
Że odmierzanie czasu samo w sobie jest oddzielne. Taka jakis obiekt, ktory informuje, ze minela sekunda itp. i ma ustawiony jakis max czas.

Inne komponenty sluchaja na to i jak timer powie, ze minela sekunda to np. ten text się zmienia, a miecz się skraca.